src/Storages/System/StorageSystemPartsColumns.cpp

```diff
@@ -94,7 +96,10 @@ void StorageSystemPartsColumns::processNextStorage(
             column_info.default_kind = toString(column.default_desc.kind);
             column_info.default_expression = queryToString(column.default_desc.expression);
         }
-
+        if (column.codec)
+        {
+            column_info.codec = queryToString(column.codec);
```

Comment:
But we can change compression codecs with `ALTER`  queries. And obviously some parts can be compressed with old codec and some with new. We must handle it.